text/1003-cli-naming.rst

```diff
@@ -19,110 +19,238 @@ specified.
 Motivation
 ==========
 
-The current naming scheme of EdgeDB command line tools is inconsistent.  Both
-``<group> <command>`` and ``<action>-<category>`` schemes are used, e.g.::
+Since the CLI commands are likely to land in third-party documentation
+and automation, we want to ensure the design behind them is consistent
+and natural to the user.
 
-  $ edgedb -h
-  <...>
-  SUBCOMMANDS:
-    alter-role
-    configure
-    create-database
-    create-superuser-role
-    describe
-    drop-role
-    dump
-    help
-    list-aliases
-    list-casts
-    list-databases
-    list-indexes
-    list-modules
-    list-object-types
-    list-ports
-    list-roles
-    list-scalar-types
-    query
-    restore
-    server
 
-and::
+Guidelines for future CLI commands
+==================================
 
-  $ edgedb server -h
-  <...>
-  SUBCOMMANDS:
-    help       Prints this message or the help of the given subcommand(s)
-    init       Initialize a new server instance
-    install    Install edgedb-server
-    restart    Restart an instance
-    start      Start an instance
-    status     Status of an instance
-    stop       Stop an instance
+* For commands operating on a single server, especially commands around
+  a single database within a single server, prefer ``edgedb <action>``
+  and ``edgedb <action>-<object>`` over ``edgedb <group> <command>``;
+
+* For commands allowing maintenance of many servers, use
+  ``edgedb server <action>`` and ``edgedb server <action>-<object>``;
+
+* For interactive output, ``--help`` that exceeds the height of the
+  current terminal should be shown in a pager like ``less``.
+
+* When using the ``<action>-<object>`` scheme, if the action in question
+  would be the literal word "show", it can be omitted, for example:
+  ``edgedb migration-log``.
+
+* Sub-command descriptions need to be as informative as possible.
+  Repeating the name of the sub-command is not very useful without
+  highlighting some details of the action.
+
+
+Specific Concerns
+=================
+
+During discussions about the CLI, a number of specific issues were
+raised. They are addressed below for reference.
+
+Ballooning ``--help`` output
+----------------------------
 
-Furthermore, RFC 1000 adds more top-level subcommands, such as:
+**Concern**: With more and more functionality exposed to the CLI, the
+``<action>-<category>`` approach has the potential to balloon the output
+of ``edgedb -h`` to more than 100 lines, making it harder to read.
 
-* ``edgedb migrate``,
-* ``edgedb create-migration``,
-* ``edgedb show-status``.
+**Decision**: a long but flat ``--help`` provides better discoverability
```

Comment:
ACTIONABLE ITEM 1: invoke a pager on long --help.

Reply:
+1 for the pager.

Also, I remember we discussed this on one of the calls: how about we show short help for `-h` and long, possibly paginated help for `--help`? That's what `git` does, btw, and I quite like it.